mithril-stm/src/stm.rs

```diff
@@ -304,13 +311,40 @@ impl StmInitializer {
         if my_index.is_none() {
             return Err(RegisterError::UnregisteredInitializer);
         }
+        Ok(my_index)
+    }
 
-        Ok(StmSigner {
-            mt_index: my_index.unwrap(),
+    /// Returns the StmSigner of the initializer for given index.
+    pub fn new_signer(self, index: Index) -> StmSigner {
+        StmSigner {
+            mt_index: index,
             stake: self.stake,
             params: self.params,
             sk: self.sk,
             vk: self.pk.vk,
+        }
+    }
+
+    /// Build the `avk` for the given list of parties.
+    ///
+    /// Note that if this StmInitializer was modified *between* the last call to `register`,
+    /// then the resulting `StmSigner` may not be able to produce valid signatures.
+    ///
+    /// Returns an `StmSigner` specialized to
+    /// * this `StmSigner`'s ID and current stake
+    /// * this `StmSigner`'s parameter valuation
+    /// * the `avk` as built from the current registered parties (according to the registration service)
+    /// * the current total stake (according to the registration service)
+    /// # Error
+    /// This function fails if the initializer is not registered.
+    pub fn new_signer_avk<D: Digest + Clone>(
```

Comment:
It probably makes sense to keep the `new_signer_avk` function as it was previously defined, and add a new one in the `StmSigner` struct for the case where we don't need the registration.